python/app/shotgun_console.py

```diff
@@ -42,15 +42,17 @@ def __init__(self, parent=None):
 
         super(ShotgunPythonConsoleWidget, self).__init__(parent)
 
-        engine = current_engine()
-        self._settings_manager = settings.UserSettings(sgtk.platform.current_bundle())
+        # Use the engine passed in or default to the current engine
+        engine = engine or current_engine()
```

Comment:
Let's just set this to `self._engine`, then you won't need to pass it around to get access to it from `_get_sg_globals`, since it'll just be able to get it from `self`.

Reply:
Sounds good - fixed